### PR TITLE
fix: fix periodic donation do not store pay_method field

### PR DESCRIPTION
**Pull request by Aylie-Chou**
This patch add gorm config to store pay_method field.

### CODE REVIEW

File: models/donation.go [hunk 1]
```diff
@@ -143,25 +143,25 @@ type PeriodicDonation struct {
...
+	Notes            string     `gorm:"type:varchar(100)" json:"notes"`
+	OrderNumber      string     `gorm:"type:varchar(50);not null" json:"order_number"`
+	SendReceipt      string     `gorm:"type:ENUM('yearly', 'no', 'no_receipt', 'digital_receipt_by_year', 'paperback_receipt_by_year');default:'no_receipt'" json:"send_receipt"`
+	Status           string     `gorm:"type:ENUM('to_pay','paying','paid','fail','stopped','invalid');not null" json:"status"`
+	ToFeedback       null.Bool  `gorm:"type:tinyint(1);default:1" json:"to_feedback"`
+	UpdatedAt        time.Time  `json:"updated_at"`
+	UserID           uint       `gorm:"type:int(10) unsigned;not null" json:"user_id"`
+	IsAnonymous      null.Bool  `gorm:"type:tinyint(1);default:0" json:"is_anonymous"`
+	AutoTaxDeduction null.Bool  `gorm:"type:tinyint(1)" json:"auto_tax_deduction"`
+	PayMethod        string     `gorm:"type:ENUM('credit_card','line','apple','google','samsung')" json:"pay_method"`
```
Comment (taylrj) [inline models/donation.go:166]: Question: Do we need to create a database index for `PayMethod` such as `PayMethod` in `PayByPrimeDonation` ?

Comment (Aylie-Chou) [reply models/donation.go:166]: For now, periodic donation has only one pay method, thus index on pay method has no benefit.
If we want to add pay method in the future, there're more than index to do:
- controller/donation should support mutiple pay method (currently we use credit card in all place)
- pay method index
  - model/donation gorm config add index
  - migration file add index
  - migration file update pay method column (for data before this patch)
- add pay method test case

In this patch, we only deal with storing pay method into database, and I think futhur work could be done when we like to add other pay method.
What do say about it ?

Comment (taylrj) [reply models/donation.go:166]: It's clear, thanks for explaining that to me.

**Review by taylrj — APPROVED** [review]
LGTM

**Review by howar31 — APPROVED** [review]
LGTM

**Pull request merged**